syslog-ng/README.md

```diff
@@ -27,12 +27,14 @@ sudo docker run -it -p 514:514/udp -p 601:601 --name syslog-ng balabit/syslog-ng
 ```
 
 ## Using custom syslog-ng configuration
-You can override the default configuration by mounting a configuration file under `/etc/syslog-ng/syslog-ng.conf`:
+You can extend the default configuration by mounting a configuration folder under `/etc/syslog-ng/conf.d/syslog-ng.conf`:
 
 ```bash
-sudo docker run -it -v "$PWD/syslog-ng.conf":/etc/syslog-ng/syslog-ng.conf balabit/syslog-ng:latest
+sudo docker run -it -v "$PWD/conf.d":/etc/syslog-ng/conf.d balabit/syslog-ng:latest
```

Comment:
I was thinking about this, and introducing `conf.d` might be a little confusing.

I understand we call `include "conf.d/*"` in `syslog-ng.conf`, but there are also default drivers provided in `syslog-ng.conf` too. Let's say someone is not fine with default-network-drivers, then they still need to override `syslog-ng.conf`. But if they override `syslog-ng.conf`, they are also free to choose to add a config file that includes from `conf.d` or any other directory with different name. Or they can put everything into a single configuration file.

One resolution to this would be is to move the default config part into the `conf.d` directory, for example into a `conf.d/default.conf` file. But then the extension usecase will not work.

You could resolve this in the documentation:
What if you maintain the original comment too?
`You can override [...] under /etc/syslog-ng/syslog-ng.conf`.

And add this besides it:
`You can extend [...]  "$PWD/conf.d":/etc/syslog-ng/conf.d [...] `

Or maybe we should just drop this `conf.d` include. It might confuse inexperienced users. However,  those who are experienced can do this themselves. I am not entirely convinced that the complexity worth it.

Reply:
I'd like to put in a vote for being able to extend config without having to override the default config. I think this is actually especially useful for inexperienced users (who I am one of). So here’s my use case and reasoning: 

In our production environment our docker app relies on a number of services including syslog. I'm trying to use docker-compose to replicate a production environment as close as possible. In the development environment I want the output of the app  to show up in docker-composes output (Which simply prints everything each container sends to stdout and stderr).  However I want to try and avoid modifying app because I want that to have as few differences between production and development as possible to reduce errors which only show up in production. So to do that I want to add a docker-compose service to replicate this syslog service. So far so good. However syslog needs to output to stdout to show up in docker-compose’s output. To do that I need to modify syslog-ng’s config. However I’m no syslog expert (quite the opposite) and I don’t want to be responsible for maintaining sensible defaults. When the syslog image gets updated along with its config file I want to be using the new config version and just include my modifications.  Ideally I would just be able to mount a single config file into the conf.d dir which would do this:

````
source s_docker_syslog { file("/var/log/messages"); };
destination d_docker_syslog { file("/dev/stdout"); };
log {
    source(s_docker_syslog);
    destination(d_docker_syslog);

};
````

However at the moment in order for that to be picked up I have to override syslog-ng.conf to add a `@include 'conf.d/*.conf'` statement. Thus negating the whole point add my modifications in a separate file since now I also am responsible for maintaining `@include 'conf.d/*.conf'`. When it comes to increasing complexity for new users I don’t see any downside in adding `@include 'conf.d/*.conf'` to syslog-ng.conf without mentioning it in the readme (although I think with some thoughtfully wording it could be mentioned in the readme without making things more confusing). That way someone like me who opens the config to modify it would see that I could just mount a seperate config file into the container and just the changes I want included.

I actually came here to submit a pull request to just add the include line. But then I came across this request which includes that plus a whole bunch of other changes. I can see that this has stalled though. If you would be amenable to adding an include statement and @rfaircloth-splunk doesn't have a chance to respond to this I'll make a new pull request for just the include statement and that way when @rfaircloth-splunk does get a chance to work on this again they can easily rebase of that.

Reply:
Hello @callumgare,

Sorry for the late response.

I understand your reasoning. I agree with adding a `conf.d` include pragma, but I'd do that in the last line, just like in our Debian packaging:
https://github.com/syslog-ng/syslog-ng/blob/0845f82f0fa07da36f0ebd6de49cdfeaac651015/packaging/debian/syslog-ng.conf#L161

I think it's a good idea to mention the `conf.d` expansibility in the readme, but please do not remove/replace the line `You can override the default configuration by mounting a configuration file under /etc/syslog-ng/syslog-ng.conf`, because that is our original and main way of editing the configuration. The `conf.d` extension can be mentioned right after this sentence.

Please answer @furiel's questions about `/dev/xconsole` and other config changes, so we can merge your PR.

Thanks for your contribution.